lib/sign_in/idme/service.rb

```diff
@@ -58,61 +59,50 @@ def user_info(token)
 
       private
 
-      def redirect_uri
-        case type
-        when 'idme'
-          config.redirect_uri
-        when 'dslogon'
-          config.dslogon_redirect_uri
-        when 'mhv'
-          config.mhv_redirect_uri
-        end
-      end
-
-      def idme_normalized_attributes(user_info)
+      def standard_attributes(user_info, credential_level)
+        loa_current = ial_to_loa(credential_level.current_ial)
+        loa_highest = ial_to_loa(credential_level.max_ial)
         {
           uuid: user_info.sub,
           idme_uuid: user_info.sub,
-          loa: { current: user_info.level_of_assurance, highest: user_info.level_of_assurance },
+          loa: { current: loa_current, highest: loa_highest },
+          sign_in: { service_name: config.service_name, auth_broker: SignIn::Constants::Auth::BROKER_CODE },
+          csp_email: user_info.email,
+          authn_context: type
+        }
+      end
+
+      def idme_attributes(user_info)
+        {
           ssn: user_info.social&.tr('-', ''),
           birth_date: user_info.birth_date,
           first_name: user_info.fname,
-          last_name: user_info.lname,
-          csp_email: user_info.email,
-          sign_in: { service_name: config.service_name, auth_broker: SignIn::Constants::Auth::BROKER_CODE },
-          authn_context: type
+          last_name: user_info.lname
         }
       end
 
-      def dslogon_normalized_attributes(user_info)
+      def dslogon_attributes(user_info)
         {
-          uuid: user_info.sub,
-          idme_uuid: user_info.sub,
-          loa: { current: user_info.level_of_assurance, highest: user_info.level_of_assurance },
           ssn: user_info.dslogon_idvalue&.tr('-', ''),
           birth_date: user_info.dslogon_birth_date,
           first_name: user_info.dslogon_fname,
           middle_name: user_info.dslogon_mname,
           last_name: user_info.dslogon_lname,
-          edipi: user_info.dslogon_uuid,
-          csp_email: user_info.email,
-          sign_in: { service_name: config.service_name, auth_broker: SignIn::Constants::Auth::BROKER_CODE },
-          authn_context: type
+          edipi: user_info.dslogon_uuid
         }
       end
 
-      def mhv_normalized_attributes(user_info)
+      def mhv_attributes(user_info)
         {
-          uuid: user_info.sub,
-          idme_uuid: user_info.sub,
-          loa: { current: user_info.level_of_assurance, highest: user_info.level_of_assurance },
           mhv_correlation_id: user_info.mhv_uuid,
-          mhv_icn: user_info.mhv_icn,
-          sign_in: { service_name: config.service_name, auth_broker: SignIn::Constants::Auth::BROKER_CODE },
-          authn_context: type
+          mhv_icn: user_info.mhv_icn
         }
       end
 
+      def ial_to_loa(ial)
+        ial == IAL::TWO ? LOA::THREE : LOA::ONE
```

Comment:
Just want to check that this isn't a typo and if IAL::TWO, should be set to LOA::THREE?